### ISSUE TITLE
`Cell` should allow to cast to `PolyData` for 2D cells

### ISSUE THREAD

**Issue by MatthewFlamm** [pyvista/pyvista#4291]
### Describe the feature you would like to be added.

https://github.com/pyvista/pyvista/discussions/4286#discussion-5093407 shows that some users would like to interact with faces as `PolyData`.  We might be able to form the 1 cell PolyData mesh similar to how `cast_to_unstructured_grid` works for `Cell`:

https://github.com/pyvista/pyvista/blob/965618ed9f71aa69caf1eee973e2325627e178d1/pyvista/core/cell.py#L168-L172

### Pseudocode or Screenshots

```python
Insert code here
```
Or drag your screenshot here!

**Comment by banesullivan** [comment]
I agree though, there should be a `cast_to_polydata()` method that works if the dimensionality supports it

**Comment by MatthewFlamm** [comment]
The problem I have with checking like this:

```py
e = cell.cast_to_unstructured_grid()
if cell.dimension < 3:
    e = e.extract_surface()
```

is that you can have valid 2D cells in an `UnstructuredGrid` that will be transformed to some other cell type in `extract_surface` to be `PolyData`.  We should let users explicitly decide that they want to make this conversion using the `cast_to_poly_data()`  We should raise an error if the cell is not 2D in this case.
